src/tracks/SampleTrack.cpp

```diff
@@ -667,6 +668,47 @@ void SampleTCOView::reverseSample()
 
 
 
+//! Split this TCO
+bool SampleTCOView::splitTCO( QMouseEvent * me )
+{
+	m_tco->setMarkerEnabled( false );
+
+	const int relativePixelPos = me->pos().x();
+	const float ppb = m_trackView->trackContainerView()->pixelsPerBar();
+	TimePos splitPos = relativePixelPos * TimePos::ticksPerBar() / ppb;
+
+	if ( !unquantizedModHeld(me) )
+	{
+		splitPos = quantizeMarkerPos( splitPos, me->modifiers() & Qt::ShiftModifier );
+	}
```

Comment:
I was thinking here, since we have the position of the marker stored on `m_marker`, couldn't we use it on this method to avoid calling the `unquantizedModHeld` and `quantizeMarkerPos` methods? Since the marker position visually represents the place where the splitting should happen, it should be accurate on the moment this method is called.

That would save a couple method calls and maybe even make it easier to move away from using the mouse event, since that conditional was the trickier part to get around without using the event.

Reply:
See comments about this same idea on previous version:

> You'd think that the result from knifeMarkerPos( me ) could be converted back to a MidiTime and used for the actual cut itself, but for some reason the shift/default behavior ends up inverted if you do. Adding a boolean switch to flip the two still left some weirdness, so I moved the code that decides the position back into mouseReleaseEvent.

https://github.com/LMMS/lmms/pull/4998#issuecomment-515694258

https://github.com/LMMS/lmms/pull/4998#issuecomment-515707007

Reply:
That's odd, I wonder why that didn't behave properly. I know we are in a bit of a schedule, but you mind if I investigate this a little? If I don't figure out soon enough why using `m_marker` didn't work we can go with the current method.